Dockerfile

```diff
@@ -1,3 +1,15 @@
+FROM --platform=$BUILDPLATFORM node:21.6-alpine3.18 AS client-builder
+WORKDIR /ui
+# cache packages in layer
+COPY ui/package.json /ui/package.json
+COPY ui/package-lock.json /ui/package-lock.json
+RUN --mount=type=cache,target=/usr/src/app/.npm \
+    npm set cache /usr/src/app/.npm && \
+    npm ci
+# install
+COPY ui /ui
+RUN npm run build
+
 FROM node:18-alpine
```

Comment:
Maybe add --platform=$BUILDPLATFORM

Reply:
@ZoranPandovski I am using the recommended Dockerfile for a React app as from Docker as given [here](https://docs.docker.com/desktop/extensions-sdk/build/frontend-extension-tutorial/). It does not use this flag in the last stage of the build (because I think the `alpine` image might be platform agnostic). Do you still want me to add it?

By the way, I change the base image used for the last stage from a `node:alpine` image to an `alpine` image as this is what has been used in the above.